### PR TITLE
adding migrations implementation #23

### PR DESCRIPTION
**Pull request by gonziesc**
## Summary
- Implemented migrations 
- Required to run migrations before executing the app
- Run migrations with "npm run migrations"
- Added environment variable

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -5,34 +5,35 @@ const express = require('express'),
...
   orm = require('./app/orm'),
-  errors = require('./app/middlewares/errors');
+  errors = require('./app/middlewares/errors'),
+  migrationsManager = require('./migrations/migrations');
 
+const app = express();
+const port = config.common.port || 8080;
 const init = () => {
-  const app = express();
-  const port = config.common.port || 8080;
```
Comment (mishuagopian) [inline app.js]: I'd move these two inside the `init` function again

File: app.js [hunk 2]
```diff
@@ -5,34 +5,35 @@ const express = require('express'),
...
   module.exports = app;
-
```
Comment (mishuagopian) [inline app.js]: readd

File: app.js [hunk 3]
```diff
@@ -5,34 +5,35 @@ const express = require('express'),
...
   // Client must send "Content-Type: application/json" header
   app.use(bodyParser.json());
   app.use(bodyParser.urlencoded({ extended: true }));
 
-  if (config.environment !== 'testing') {
+  if (!config.isTesting) {
     morgan.token('req-params', (req) => req.params);
     app.use(morgan('[:date[clf]] :remote-addr - Request ":method :url" with params: :req-params. Response status: :status.'));
   }
-
```
Comment (mishuagopian) [inline app.js]: readd

File: app.js [hunk 4]
```diff
@@ -5,34 +5,35 @@ const express = require('express'),
...
-  orm.init(app).then(() => {
+  const initializer = Promise.resolve();
+  initializer.then(() => {
```
Comment (mishuagopian) [inline app.js]: This may look cleaner this way:
```node
Promise.resolve().then(() => {
```

File: app.js [hunk 5]
```diff
@@ -5,34 +5,35 @@ const express = require('express'),
...
+    if (!config.isTesting) {
+      return migrationsManager.init();
+    }
+  }).then(() => orm.init(app)).then(() => {
     routes.init(app);
-
```
Comment (mishuagopian) [inline app.js]: readd

File: app.js [hunk 6]
```diff
@@ -5,34 +5,35 @@ const express = require('express'),
...
     app.use(errors.handle);
     app.use(rollbar.errorHandler(config.common.rollbar.accessToken, {
       enabled: !!config.common.rollbar.accessToken,
       environment: config.environment
     }));
-
```
Comment (mishuagopian) [inline app.js]: readd

File: app.js [hunk 7]
```diff
@@ -5,34 +5,35 @@ const express = require('express'),
...
     app.listen(port);
     console.log(`Listening on port: ${port}`); // eslint-disable-line
   }).catch(console.log); // eslint-disable-line
 };
-
```
Comment (mishuagopian) [inline app.js]: readd

File: migrations/migrations.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+const Umzug = require('umzug'),
+  Sequelize = require('sequelize'),
+  config = require('../config/config').config,
+  orm = require('../app/orm'),
+  errors = require('../app/errors');
+
+exports.init = () => {
```
Comment (mishuagopian) [inline migrations/migrations.js]: I'd name this `check`. Resulting in `migrations.check()` when using it.

File: migrations/migrations.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    },
+    migrations: {
+      params: [
+        db.getQueryInterface(),
+        db.constructor,
+        function () {
+          throw new Error('Migration tried to use old style "done" callback.upgrade');
+        }
+      ],
+      path: '././migrations/migrations',
```
Comment (mishuagopian) [inline migrations/migrations.js]: `./migrations/migrations`

File: migrations/migrations/booksmigrations.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
+module.exports = {
```
Comment (mishuagopian) [inline migrations/migrations/booksmigrations.js]: Migrations should have the creation timestamp in its name. Did you delete it?

File: migrations/migrations/usermigration.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
+module.exports = {
```
Comment (mishuagopian) [inline migrations/migrations/usermigration.js]: Same thing about the timestamp

File: migrations/migrations/booksmigrations.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+  up: (queryInterface, Sequelize) => {
+    queryInterface.createTable('book', {
```
Comment (MimoMoreno) [inline migrations/migrations/booksmigrations.js]: Should return de promise like this: `return queryInterface.createTable(...)`

File: migrations/migrations/usermigration.js [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+  up: (queryInterface, Sequelize) => {
+    queryInterface.createTable('user', {
```
Comment (MimoMoreno) [inline migrations/migrations/usermigration.js]: Should return the promise

File: migrations/config.js [hunk 14]
```diff
@@ -0,0 +1,5 @@
+const config = require('../config/config').config;
```
Comment (mishuagopian) [inline migrations/config.js]: This has changed. Require `config` as the following:
```es6
const config = require('./../config');
```

File: migrations/migrations.js [hunk 15]
```diff
@@ -0,0 +1,33 @@
+const Umzug = require('umzug'),
+  Sequelize = require('sequelize'),
+  config = require('../config/config').config,
```
Comment (mishuagopian) [inline migrations/migrations.js]: same

File: migrations/config.js (hunk 14 above)
Comment (gonziesc) [inline migrations/config.js]: It's   `config = require('./../config/config').config`

**Pull request merged**